pkg/controller/resourcequota/resource_quota_controller.go

```diff
@@ -292,16 +292,22 @@ func (rq *Controller) worker(queue workqueue.TypedRateLimitingInterface[string])
 // Run begins quota controller using the specified number of workers
 func (rq *Controller) Run(ctx context.Context, workers int) {
 	defer utilruntime.HandleCrashWithContext(ctx)
-	defer rq.queue.ShutDown()
-	defer rq.missingUsageQueue.ShutDown()
 
 	logger := klog.FromContext(ctx)
-
 	logger.Info("Starting resource quota controller")
-	defer logger.Info("Shutting down resource quota controller")
+
+	var wg sync.WaitGroup
+	defer func() {
+		logger.Info("Shutting down resource quota controller")
+		rq.queue.ShutDown()
```

Comment:
Ditto LIFO order.

Reply:
I did align the order in the end, as you wished.